onnxruntime/contrib_ops/cuda/bert/rotary_embedding.cc

```diff
@@ -57,10 +57,10 @@ Status RotaryEmbedding<T>::ComputeInternal(OpKernelContext* context) const {
 
   Tensor* output = context->Output(0, input->Shape());
 
-  if (parameters.sequence_length > parameters.max_sequence_length) {
-    // Launch update_cos_sin_cache kernel with scale
-    ORT_NOT_IMPLEMENTED("Updating cos_cache and sin_cache in RotaryEmbedding is not currently supported");
-  }
+  // if (parameters.sequence_length > parameters.max_sequence_length) {
```

Comment:
If this check is removed, need compute rotary without cache

Reply:
The correct way is to check the max_value in position_ids, however, this will slow down the kernel speed along with D2H operations.
Input_ids can be multi-batch packed together with which setence has different length, and the position_ids will truely describe its components.